src/main/java/com/elixr/poc/rest/response/PurchaseExceptionHandler.java

```diff
@@ -0,0 +1,33 @@
+package com.elixr.poc.rest.response;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@ControllerAdvice
+public class PurchaseExceptionHandler {
+    private PurchaseErrorResponse errorResponse;
```

Comment:
This instance variable is not needed here, you can directly declare it in your method